src/core/qgsofflineediting.cpp

```diff
@@ -817,7 +831,7 @@ void QgsOfflineEditing::applyAttributeValueChanges( QgsVectorLayer* offlineLayer
   for ( int i = 0; i < values.size(); i++ )
   {
     QgsFeatureId fid = remoteFid( db, layerId, values.at( i ).fid );
-
+    QgsDebugMsg( QString( "Offline changeAttributeValue %1 = %2" ).arg( QString( attrLookup[ values.at( i ).attr ] ), values.at( i ).value ) );
```

Comment:
No problem to remove that debug message.
Do you mean that the other debug messages should become `QgsDebugMsgLevel` ? What level do you think would be appropriate?

I'm not sure that thing are really fixed for WFS, this test passes locally but I'd like to add another test for the case:
- go offline
- edit
- delete the first or the last feature with an online WFS call
- sync

I fear that the fid<->id mapping will non be correct anymore, don't we really need a WFS fid<->id mapping stored in the offline DB?

Also, tests cases for offline INSERT and DELETE are still missing.